var/spack/repos/builtin/packages/intel-llvm/package.py

```diff
@@ -0,0 +1,49 @@
+# Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class IntelLlvm(CMakePackage):
+    """Intel's version of the LLVM compiler.
+    """
+
+    homepage = 'https://github.com/intel/llvm'
+    git = 'https://github.com/intel/llvm.git'
+
+    family = 'compiler'
+
+    version('sycl', branch='sycl')
+
+    depends_on('cmake@3.4.3:', type='build')
+
+    # It doesn't seem possible to use != in a conflicts statement
+    # conflicts('target = x86_64',
+    #            msg='Intel LLVM compiler currently only works for x86')
+
+    def setup_environment(self, spack_env, run_env):
+        spack_env.append_flags('CXXFLAGS', self.compiler.cxx11_flag)
+
+        if '+clang' in self.spec:
+            run_env.set('CC', join_path(self.spec.prefix.bin, 'clang'))
+            run_env.set('CXX', join_path(self.spec.prefix.bin, 'clang++'))
+
+    root_cmakelists_dir = join_path('llvm')
+
+    def cmake_args(self):
+
+        cmake_args = []
+
+        cmake_args.extend([
```

Comment:
Why create an empty list, extend it, then return it, when you could just return a list?
```suggestion
        return [
```

Reply:
I disagree with this. I like to write packages this way initially because the arguments almost always require logic at some point. Since I have to look up syntax for every language I program in, it just makes it easier for someone in the future to work on in my opinion.